nix-lib/default.nix

```diff
@@ -109,6 +109,40 @@ let
     in
       builtins.filterSource (file: _: ! (hasAnyPrefix file)) path;
 
+  # Build nix expression of dependencies based on given `yarnLock`
+  # and directly `callPackage` it. Using this can avoid the need to
+  # check in a generated nix expression. The resulting attrSet can
+  # be used as input for `buildNodeDeps`:
+  #
+  # ```
+  # buildNodeDeps (buildCallDeps { yarnLock = ./yarn.lock; })
+  # ```
+  buildCallDeps = { name ? "npm-deps.nix", yarnLock }:
```

Comment:
Probably should name `buildCallDeps` something different

Reply:
Yeah, the name is not great.

`haskellPackages` uses `callCabal2nix`, so let’s use `callYarn2Nix`?

Reply:
Maybe a bit unclear in distinction to the `--template` version? `callYarnLock2Nix` too cumbersome on the other hand.